_posts/2021/2021-03-31-hmat_memkind.md

```diff
@@ -0,0 +1,172 @@
+
+The following table contains a comparison between hardware and expected memory property associated with it:
+
+| Physical medium      | Memory property |
+| -------------------- | ----------------|
+| DRAM                 | Latency         |
+| [MCDRAM][MCDRAM]     | Bandwidth       |
+| PMem                 | Capacity        |
+
+<figure class="image">
+  <img src="/assets/hazelcast_3.png" alt="PMEM KMEM-DAX">
+  <figcaption>Figure 3. Memory architecture overview based on physical medium.</figcaption>
+</figure>
+
+And presented same view in different form:
+
+<figure class="image">
+  <img src="/assets/memkind_memory_view.png" alt="CAPACITY LATENCY VIEW">
+  <figcaption>Figure 4. Memory architecture overview based on memory property.</figcaption>
+</figure>
+
+To follow this belief we extend memkind library with additional semantics.
+
+### Prerequisites
+
+[ACPI-6.2][ACPI-doc] extends the ACPI standard about Heterogeneous Memory Attribute
+Table (HMAT) table. HMAT provides a way for the firmware to propagate to Operating System memory hardware
+characteristics - in details, it describes bandwidth and latency from the  **initiator of memory requests**
+(a processor or an accelerator) to any **memory target**.
+
+Besides support for HMAT in hardware, support is also required on the Linux kernel layer*:
+
+{% highlight console %}
+$ make nconfig
+	Power management and ACPI options --->
+		[*] ACPI (Advanced Configuration and Power Interface) Support --->
+			-*-   NUMA support
+			[*]     ACPI Heterogeneous Memory Attribute Table Support
+{% endhighlight %}
+
+* Option **CONFIG_ACPI_HMAT** is available since version 5.5
+
+To utilize information provided by the OS memkind use [hwloc library][hwloc] therefore, it is additional mandatory dependency
+if you want to use new features offer by memkind.
+
+### Memory kind summary
+
+With [memkind 1.11.0 release][memkind-release] comes support for previously mentioned memory property semantics in following API:
+
+| New API                                       | Associated Memory property |
+| ----------------------------------------------| -------------------------- |
+| **MEMKIND_HIGHEST_CAPACITY**	                |   Capacity                 |
+| **MEMKIND_HIGHEST_CAPACITY_PREFERRED**		    |   Capacity                 |		
+| **MEMKIND_HIGHEST_CAPACITY_LOCAL**            |   Capacity                 |
+| **MEMKIND_HIGHEST_CAPACITY_LOCAL_PREFERRED**  |   Capacity                 |
+| **MEMKIND_LOWEST_LATENCY_LOCAL**              |   Latency                  |
+| **MEMKIND_LOWEST_LATENCY_LOCAL_PREFERRED**    |   Latency                  |
+| **MEMKIND_HIGHEST_BANDWIDTH_LOCAL**           |   Bandwidth                |
+| **MEMKIND_HIGHEST_BANDWIDTH_LOCAL_PREFERRED**	|   Bandwidth                |
+
+The new semantics based on comparison specific memory property between different NUMA Nodes presented on the platform and choosing
```

Comment:
The new semantics are based on a comparison between .. on different NUMA nodes on the platform.
The new semantics choose the best .. on different NUMA nodes available on the platform.

?